src/index.ts

```diff
@@ -72,20 +73,14 @@ export default class Kumulos {
         }
 
         if (this.context.hasFeature(SDKFeature.DDL)) {
-            if (!isMobile()) {
-                console.info(
-                    'DdlManager: DDL feature support only available on mobile devices.'
-                );
-                return;
-            }
-
-            this.ddlManager = new DdlManager(this.context, this.rootFrame);
+            this.initializeDDLFeature();
         }
     }
 
     private initializePushFeature() {
         trackOpenFromQuery(this.context);
         registerServiceWorker(this.context.serviceWorkerPath);
+        this.observePermissionStatus();
```

Comment:
Should we observe before SW reg?

Although both async, wonder if any potential races we could 'miss' some change?

Probably not.

Reply:
We are not awaiting the registration so I dont think it matters